### PR TITLE
The Bassed Pro Shop

### PR DESCRIPTION
**Pull request by minervatbd**

### CODE REVIEW

File: client.py [hunk 1]
```diff
@@ -400,15 +411,15 @@
 	ewcfg.cmd_arrest: ewcmd.arrest,
 }
 
-debug = False
+debug = True
```
Comment (magicksRampage) [inline client.py:414]: turn debug mode off

File: client.py [hunk 2]
```diff
@@ -400,15 +411,15 @@
...
 while sys.argv:
 	if sys.argv[0].lower() == '--debug':
 		debug = True
 
 	sys.argv = sys.argv[1:]
 
 # When debug is enabled, additional commands are turned on.
-if debug == True:
+if debug == False:
```
Comment (magicksRampage) [inline client.py:422]: this makes no sense

File: ewcfg.py [hunk 3]
```diff
@@ -470,9 +472,17 @@
...
 cmd_cast = cmd_prefix + 'cast'
 cmd_reel = cmd_prefix + 'reel'
+cmd_grill = cmd_prefix + 'grill'
 cmd_appraise = cmd_prefix + 'appraise'
+cmd_appraise_alt1 = cmd_prefix + 'measure'
 cmd_barter = cmd_prefix + 'barter'
 cmd_embiggen = cmd_prefix + 'embiggen'
+cmd_enter = cmd_prefix + 'enter'
+cmd_store = cmd_prefix + 'store'
+cmd_take = cmd_prefix + 'take'
```
Comment (magicksRampage) [inline ewcfg.py:482]: enter, store and take are too generic as commands. please rename them something more specific to fishing

File: ewfish.py [hunk 4]
```diff
@@ -106,7 +112,62 @@ def deal(self):
...
+					ewcfg.col_id_user,
+					ewcfg.col_id_tourney,
+					ewcfg.col_id_fish,
+					ewcfg.col_float_size,
+					ewcfg.col_rank_int,
+				), (
+					id_server,
+					id_user,
+				)
+			)
```
Comment (magicksRampage) [inline ewfish.py:153]: you're not doing anything with this data

File: ewfish.py [hunk 5]
```diff
@@ -364,7 +430,7 @@ def gen_bite_text(size):
 			if item_sought:
 				item = EwItem(id_item = item_sought.get('id_item'))
 
-				if item.item_type == ewcfg.it_food:
+				if item.item_type == ewcfg.it_food and item.item_props["acquisition"] != ewcfg.acquisition_fishing:
```
Comment (magicksRampage) [inline ewfish.py:433]: why not? lol

File: ewfish.py [hunk 6]
```diff
@@ -430,11 +496,11 @@ def gen_bite_text(size):
 			bite_text = gen_bite_text(fisher.current_size)
 			
 			# User has a 1/10 chance to get a bite
-			fun = 100
+			fun = 10
 
 			if fisher.bait == True:
 				# Bait attatched, user has a 1/7 chance to get a bite
-				fun = 70
+				fun = 7
```
Comment (magicksRampage) [inline ewfish.py:503]: change debug values back

File: ewfish.py [hunk 7]
```diff
@@ -459,13 +525,17 @@ def gen_bite_text(size):
...
 					if bun >= 5:
 						fun -= 1
-					if bun >= 15:
-						fun -= 1
+					if bun >= 7:
+						fun -= 2
+					if bun >= 9:
+						fun -= 2
+					continue
+				elif fisher.rod_type == "stickpole":
```
Comment (magicksRampage) [inline ewfish.py:538]: are you supposed to never get a bite with this rod?

File: ewfish.py [hunk 8]
```diff
@@ -482,6 +552,11 @@ def gen_bite_text(size):
 				fisher.current_fish = ""
 				fisher.current_size = ""
 				fisher.bait = False
+				fisher.slime_type = ""
+				if fisher.rod_type == "gamblersdelight":
+					user_data.id_killer = cmd.message.author.id
+					user_data.die(cause=ewcfg.cause_suicide)
+					user_data.persist()
```
Comment (magicksRampage) [inline ewfish.py:559]: add a message to the pier channel and to the sewers to explain the cause of death to the players

File: ewfish.py [hunk 9]
```diff
@@ -348,13 +409,18 @@ def gen_bite_text(size):
...
+					fisher.rod_type = weapon.id_weapon
 
 			fisher.current_fish = gen_fish(market_data, cmd, has_fishingrod)
 			fisher.fishing = True
 			fisher.bait = False
 			fisher.pier = user_data.poi
+			if cmd.message.channel.name in [ewcfg.channel_afb_pier, ewcfg.channel_vc_pier, ewcfg.channel_se_pier, ewcfg.channel_ferry]:
+				fisher.slime_type = "salt"
+			else:
+				fisher.slime_type = "fresh"
```
Comment (magicksRampage) [inline ewfish.py:423]: please use the constants in ewcfg for the slime types

File: ewfish.py [hunk 10]
```diff
@@ -613,29 +736,77 @@ def gen_bite_text(size):
...
+							'size': fisher.current_size,
+							'time_expir': time_expir,
+							'acquisition': ewcfg.acquisition_fishing,
+							'value': value,
+							'date_caught': day,
+							'measured': False
+						}
+					)
+				else:
+					size = ewcfg.fish_map[fisher.current_fish].size
```
Comment (magicksRampage) [inline ewfish.py:778]: shouldn't we be using fisher.current_size here?

File: ewfish.py [hunk 11]
```diff
@@ -613,29 +736,77 @@ def gen_bite_text(size):
...
+							'recover_hunger': ewcfg.base_grill_hunger * hunger_multiplier,
+							'str_eat': ewcfg.grill_str_eat.format(size, ewcfg.fish_map[fisher.current_fish].str_name),
+							'time_expir': time.time() + ewcfg.std_food_expir,
+							'acquisition': ewcfg.acquisition_grilling,
+						}
+					)
 
-				response = "You reel in a {fish}! {flavor} You grab hold and wring {slime} slime from it. "\
-					.format(fish = ewcfg.fish_map[fisher.current_fish].str_name, flavor = ewcfg.fish_map[fisher.current_fish].str_desc, slime = str(slime_gain))
+				response = "You reel in a {fish}! {flavor} You grab hold and wring {slime} slime from it. also its {size}"\
```
Comment (magicksRampage) [inline ewfish.py:806]: probably rework this flavor text

File: ewfish.py [hunk 12]
```diff
@@ -660,6 +831,63 @@ def gen_bite_text(size):
...
+	else:
+		size_int = 160
+		size_int += (random.randint(1, 80) * rarity_multiplier) / 10000
+		size_int += (random.randint(1, 80) * rarity_multiplier) / 1000
+		size_int += (random.randint(1, 80) * rarity_multiplier) / 100
+		size_int += (random.randint(1, 80) * rarity_multiplier) / 10
+		size_int += (random.randint(1, 80) * rarity_multiplier) / 1
+	if size_int <= 0:
+		size_int = 0.0001
+	size_int = int(size_int*10000)/10000
```
Comment (magicksRampage) [inline ewfish.py:888]: why is it called size_int if it's generally not an integer?

File: ewfish.py [hunk 13]
```diff
@@ -786,194 +1059,281 @@ def gen_bite_text(size):
...
+					offer_give=id_fish
+				)
 
-			items = []
+				mixtapes = ewcfg.item_list
+				for item in mixtapes:
+					if item.acquisition == ewcfg.acquisition_lilb:
+						pass
+					else:
+						mixtapes.remove(item)
```
Comment (magicksRampage) [inline ewfish.py:1086]: the item_list is assigned by reference. you're permanently removing items from the global item list here

File: ewfish.py [hunk 14]
```diff
@@ -786,194 +1059,281 @@ def gen_bite_text(size):
...
+			else:
+				value = int(item_props['value'])
 
-						response = '\n"Well, back again I see! My offer still stands, I’ll trade ya a {} for your {}"'.format(item.str_name, name)
+				items = []
 
-					response += "\n**!accept** or **!refuse** Captain Albert Alexander's deal."
+				# Filters out all non-generic items without the current fish as an ingredient.
+				for result in ewcfg.appraise_results:
+					if result.ingredients == fish.item_props.get('id_item') or result.ingredients == "generic" and result.acquisition == ewcfg.acquisition_bartering:  # Generic means that it can be made with any fish.
```
Comment (magicksRampage) [inline ewfish.py:1161]: shouldn't it be 'id_fish'?

File: ewfish.py [hunk 15]
```diff
@@ -786,194 +1059,281 @@ def gen_bite_text(size):
...
 
-					await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+				# Filters out items of greater value than your fish.
+				for value_filter in items:
+					if value < value_filter.context:
+						items.remove(value_filter)
+					else:
+						pass
 
 				else:
```
Comment (magicksRampage) [inline ewfish.py:1173]: there is no if to this else

**Review by magicksRampage — CHANGES_REQUESTED** [review]
there's a lot here that seems questionable. I haven't gotten through all of the code yet. please do some more testing to make sure the features work as intended.

**Comment by magicksRampage** [comment]
also provide a feature summary please

**Comment by minervatbd** [comment]
once school settles down a bit ill come back to work on this.

**Pull request closed without merging**